webchat/utils/pushNotificationService.js

```diff
@@ -0,0 +1,136 @@
+(function() {
+    'use strict';
+
+    const app = angular.module('webchat');
+
+    app.service('PushNotificationService', function PushNotificationService($firebaseArray,
+        $firebaseObject, $q) {
+        /**
+         * Service responsible for send request permission
+         * to enable notifications to the user and for deal
+         * with the token resulted from this operation by saving
+         * or updating it in firebase database.
+         * Just in case the user is on a mobile device.
+         */
+        const service = this;
+
+        /**
+         * Retrieves the application instance of
+         * firebase messaging.
+         */
+        const messaging = firebase.messaging();
```

Comment:
remember that this causes a problem at safari browser

Reply:
Removed pushNotificationService as its not currently used on authentication.